File: model_dflowfm_blackbox/tests/simple_waal_kalman/plot_results_hisfile.m
 file = '.\stochModel\input_dflowfm\DFM_OUTPUT_simple_waal\simple_waal_his.nc';
 time = ncread(file,'time');
 time_days = time/(24*60*60);
 waterlevel = ncread(file,'waterlevel');

 figure(1)
 plot(time_days,waterlevel(1,:),'b')
 title('station01')
 xlabel('time (days)');
 ylabel('waterlevel');
 
 figure(2)
 plot(time_days,waterlevel(2,:))
 title('station02')
 xlabel('time (days)');
 ylabel('waterlevel');
 
 figure(3)
 plot(time_days,waterlevel(3,:))
 title('station03')
 xlabel('time (days)');
 ylabel('waterlevel');
 
 

File: model_dflowfm_blackbox/tests/estuary_kalman_pli-file/plot_results_hisfile.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
"""
    Plot results hisfile
"""
from netCDF4 import Dataset, num2date
import os
import matplotlib.pyplot as plt

# location of input file is defined relative to location of this script
ncfile = os.path.join(os.path.dirname(__file__), 'stochModel/work0/dflowfm/DFM_OUTPUT_estuary/full_estuary_his.nc')

nc_fid = Dataset(ncfile, 'r')

time_work = nc_fid.variables['time'][:]
timeunit = nc_fid['time'].units
time = num2date(time_work, timeunit)
wl = nc_fid.variables['waterlevel'][:]

fig1 = plt.figure()
plt.plot_date(time, wl[:, 0], 'b')
plt.title('station01')
plt.xlabel('time')
plt.ylabel('waterlevel')

fig2 = plt.figure()
plt.plot_date(time, wl[:, 1], 'b')
plt.title('station02')
plt.xlabel('time')
plt.ylabel('waterlevel')

fig3 = plt.figure()
plt.plot_date(time, wl[:, 2], 'b')
plt.title('station03')
plt.xlabel('time')
plt.ylabel('waterlevel')

plt.show()

 
